### ISSUE TITLE
Inter-instance command passing clobbers terminal and log.

### ISSUE THREAD

**Issue by Infini7y77** [Unvanquished/Unvanquished#672]
I have setup up as part of an openSUSE package, to install the server as a 'service', and all seems to be working with initial server config, etc.

Now, how do I connect an admin console or 'pipe' in commands (for alpha 37..) to the running daemonded? Is this available, or have to connect with a client to administer?

**Comment by illwieckz** [comment]
Since alpha 37, a new instance of unvanquished talks to the existing instance, so you can do that:

```
# fork in the background and load map plat23:
nohup unvanquished +map plat23 &

# change map to antares:
unvanquished +map antares

# talk on console:
unvanquished +say hello
```

**Comment by Amanieu** [comment]
Note that to make this work, you must pass the same -homepath parameter as you did when you started the server. This allows you to have multiple servers running, as long as they use different homepaths.

For the console log, you can use daemon.log in the server's homepath.

**Comment by Infini7y77** [comment]
Thank you.

**Comment by Infini7y77** [comment]
Sorry to bump this thread, but... few issues with the above approach.

First, the new instance used to talk to the existing running server clobbers the existing daemon.log with its output.

Second, the new instance on exit 'clobbers' the running terminal, so it looses it linefeed formating, etc (needing to run 'reset' in the terminal), after sending a command.

Also, is there a way to 'attach' to the running server, like the original instance does, but with a new instance (telnet'esque).

**Comment by illwieckz** [comment]
If the new instance clobbers the existing daemon.log, it's a bug to report and to fix (tried here and confirmed). About the terminal issue, I confirm too (and `set com_ansicolor 0` does not works anymore). About the _telnet-esque_ console, there is a `daemon-tty` binary for that but right now I tried the `daemon-tty` and it appears it disconnect itself.

Well, this behavior (another instance connects to the new one) is a very recent change (comes with 0.37.0) so there is many things to fix.

There was before a `svpipe` (that worked only on unix systems, but it's OK for you) bu I'm not able to reproduce here now. :/

You can create one pipe yourself like that:

```
mkfifo /tmp/unvpipe
tail -f /tmp/unvpipe | daemonded
```

So you can now send commands like that:

```
echo "map plat23" > /tmp/unvpipe
```

Hum, there is plenty of issues to fix:

- [ ] do not overwrite daemon.log while connecting to an existing instance
- [ ] do not break the terminal output while connecting to an existing instance
- [ ] allow `daemon-tty` to open a console on an existing server (not only send command)
- [ ] allow `set com_ansicolor 0`

**Comment by Viech** [comment]
Rewrote issue for the new problems.

**Comment by illwieckz** [comment]
@Infini7y77 
You can try that: http://pastebin.com/dQCgRsrk

@Viech: since the issue has been rewritten for new problems, perhaps reopen it?

**Comment by Amanieu** [comment]
I'm aware of the daemon.log problem and will fix it later today. I still don't quite understand what the problem is with the terminal?

Regarding com_ansiColor, the problem is that the engine starts printing stuff long before the cmdline args are parsed, so I'm not quite sure how to handle that...

Regarding "attaching" a console to an existing instance, I have thought about this in the past but it will require a lot of code to be written, so it is a longer-term project.

**Comment by Amanieu** [comment]
Log issue is fixed by 246a5fe

**Comment by Amanieu** [comment]
Terminal issue fixed by 35fc519
com_ansiColor issue fixed by b242134
